commands/chain.go

```diff
@@ -122,3 +125,62 @@ var storeStatusCmd = &cmds.Command{
 		return nil
 	},
 }
+
+var storeSetHeadCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Modify the head of the current chain.",
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("cids", true, true, "CID's of the tipset to set the chain head to."),
+	},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		headCids := make([]cid.Cid, len(req.Arguments))
+		for i, arg := range req.Arguments {
+			c, err := cid.Decode(arg)
+			if err != nil {
+				return err
+			}
+			headCids[i] = c
+		}
+		headKey := types.NewTipSetKey(headCids...)
+
+		headTs, err := GetPorcelainAPI(env).ChainTipSet(headKey)
```

Comment:
This logic, while small, belongs behind the porcelain API, not in front of it. Change `ChainSetHead` to take a `TipSetKey` rather than tipset.